From c0d9a227381350c7fa49ecabb1a8cbbfe4ff40ed Mon Sep 17 00:00:00 2001
From: Carsten Otte <cotte@de.ibm.com>
Date: Wed, 4 Jan 2012 10:25:30 +0100
Subject: [PATCH 14/42] KVM: s390: Fix return code for unknown ioctl numbers
Git-commit: c0d9a227381350c7fa49ecabb1a8cbbfe4ff40ed
References: FATE#314101,FATE#310914
Patch-mainline: Merged into kvm

This patch fixes the return code of kvm_arch_vcpu_ioctl in case
of an unkown ioctl number.

Signed-off-by: Carsten Otte <cotte@de.ibm.com>
Signed-off-by: Marcelo Tosatti <mtosatti@redhat.com>
Signed-off-by: Avi Kivity <avi@redhat.com>
Signed-off-by: Alexander Graf <agraf@suse.de>

---
 arch/s390/kvm/kvm-s390.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/arch/s390/kvm/kvm-s390.c b/arch/s390/kvm/kvm-s390.c
index e66f31a..8993c49 100644
--- a/arch/s390/kvm/kvm-s390.c
+++ b/arch/s390/kvm/kvm-s390.c
@@ -674,7 +674,7 @@ long kvm_arch_vcpu_ioctl(struct file *filp,
 		r = kvm_arch_vcpu_ioctl_initial_reset(vcpu);
 		break;
 	default:
-		r = -EINVAL;
+		r = -ENOTTY;
 	}
 	return r;
 }
-- 
1.6.0.2

